src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs

```diff
@@ -12,102 +12,36 @@
-    {
-        private readonly IMergePolicyEvaluator _mergePolicyEvaluator;
-        private readonly BuildAssetRegistryContext _context;
-        private readonly IRemoteFactory _darcFactory;
-        private readonly ILoggerFactory _loggerFactory;
-        private readonly IActionRunner _actionRunner;
-        private readonly IPullRequestPolicyFailureNotifier _pullRequestPolicyFailureNotifier;
-        private readonly IConnectionMultiplexer _redis;
-
-        /// <summary>
-        ///     Creates a new PullRequestActor
-        /// </summary>
-        /// <param name="id">
-        ///     The actor id for this actor.
-        ///     If it is a <see cref="Guid" /> actor id, then it is required to be the id of a non-batched subscription in the
-        ///     database
-        ///     If it is a <see cref="string" /> actor id, then it MUST be an actor id created with
-        ///     <see cref="PullRequestActorId.Create(string, string)" /> for use with all subscriptions targeting the specified
-        ///     repository and branch.
-        /// </param>
-        /// <param name="provider"></param>
-        public PullRequestActor(
-            IMergePolicyEvaluator mergePolicyEvaluator,
-            BuildAssetRegistryContext context,
-            IRemoteFactory darcFactory,
-            ILoggerFactory loggerFactory,
-            IActionRunner actionRunner,
-            IPullRequestPolicyFailureNotifier pullRequestPolicyFailureNotifier,
-            IConnectionMultiplexer redis)
-        {
-            _mergePolicyEvaluator = mergePolicyEvaluator;
-            _context = context;
-            _darcFactory = darcFactory;
-            _loggerFactory = loggerFactory;
-            _actionRunner = actionRunner;
-            _pullRequestPolicyFailureNotifier = pullRequestPolicyFailureNotifier;
-            _redis = redis;
-        }
-
-        public PullRequestActorImplementation Implementation { get; private set; }
-
-        public Task TrackSuccessfulAction(string action, string result)
-        {
-            return Implementation.TrackSuccessfulAction(action, result);
-        }
-
-        public Task TrackFailedAction(string action, string result, string method, string arguments)
-        {
-            return Implementation.TrackFailedAction(action, result, method, arguments);
-        }
-
-        public Task<string> RunActionAsync(string method, string arguments)
-        {
-            return Implementation.RunActionAsync(method, arguments);
-        }
-
-        public Task UpdateAssetsAsync(Guid subscriptionId, int buildId, string sourceRepo, string sourceSha, List<Asset> assets)
-        {
-            return Implementation.UpdateAssetsAsync(subscriptionId, buildId, sourceRepo, sourceSha, assets);
-        }
-    }
-
-    public abstract class PullRequestActorImplementation : IPullRequestActor, IActionTracker
+    public abstract class PullRequestActor : IPullRequestActor, IActionTracker
```

Comment:
@andriipatsula should we remove the `IActionTracker` too?

Reply:
I don't know. We may want to keep it. If I am not mistaken, it's used for audit..

Reply:
removed